docs/developer-guide/plugins-scenarios.md

```diff
@@ -233,115 +231,81 @@ However there are some pitfalls you need to know:
 In this rather contrived example we will transform HTTP response content to 
 upper case letters. 
 
-Streaming `HttpRequest` content is a byte buffer stream. The stream can be accessed
-with a `request.body()` method and its data type is `StyxObservable<ByteBuf>`.
-
-Because `toUpperCase` is non-blocking transformation we can compose it to the original
-byte stream using a `map` operator.
-
-The mapping function decodes the original byte buffer `buf` to an UTF-8 encoded
-Java `String`, and copies its upper case representation into a newly created byte buffer.
-The old byte buffer `buf` is discarded.
-
-Because `buf` is a Netty reference counted `ByteBuf`, we must take care to decrement its
-reference count by calling `buf.release()`. 
-
 ```java
 public class MyPlugin extends Plugin {
     @Override
-    public StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain) {
-        StyxObservable<ByteBuf> toUpperCase = request.body()
-                .map(buf -> {
-                    buf.release();
-                    return copiedBuffer(buf.toString(UTF_8).toUpperCase(), UTF_8);
-                });
-
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
         return chain.proceed(
                 request.newBuilder()
-                        .body(toUpperCase)
+                        .body(byteStream ->                                                              // 1
+                                byteStream.map(buf -> {                                                  // 2
+                                    String upperCase = new String(buf.content(), UTF_8).toUpperCase();   // 3
+                                    return new Buffer(upperCase, UTF_8);
+                                }))
                         .build()
         );
     }
 }    
 ```
 
-### Asynchronously transforming streaming request content
-
-Asynchronous content stream transformation is very similar to the synchronous transformation.
-The only difference is that asynchronous transformation must be composed with a `flatMap` 
-instead of `map`. Otherwise the same discussion apply. As always it is important to take care
-of the reference counting.
+A call to `request.newBuilder` opens up a new request builder that allows the request 
+to be transformed to.
```

Comment:
It looks like this sentence ended prematurely. Or should it just say:

```suggestion
to be transformed.
```
?

Reply:
Yes.